src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -587,6 +594,8 @@ case class PMethodReceivePointer(typ: PNamedOperand) extends PMethodRecvType
 
 case class PFunctionType(args: Vector[PParameter], result: PResult) extends PTypeLit with PScope
 
+case class PPredType(args: Vector[PType]) extends PTypeLit with PScope
```

Comment:
Because the predicate type now only takes types and no named parameters, you can remove the PScope.